extensibility/OnDelete/README.md

```diff
@@ -0,0 +1,12 @@
+# Overview
+Illustrates a way to support an `IDeleteAware` interface to be implemented by database classes, to recieve notifications when instances of the type is deleted.
+
+## Usage
+First copy sources from `..\Starcounter.Database.Hosting.Extensibility` to your application. We suggest you put them in a folder with the same name.
+
+Then copy `MyTransactor.cs` and `IDeleteAware.cs` to your application.
+
+Now you can implement `IDeleteAware.OnDelete` on your database types that need to take action when an instance of the type is deleted.
```

Comment:
This is ambiguous. In case of an existing application, there will be usage of `Starcounter.Database.ITransactor`, which will ignore `IDeleteAware` interface. Is it possible to overwrite `Starcounter.Database.ITransactor` service in the collection? So the application code would look exactly the same?

Reply:
> Is it possible to overwrite Starcounter.Database.ITransactor service in the collection? So the application code would look exactly the same?

Technically, yes. But I don't think we should promote that. 

IMO, it's perfectly fine to instruct any team that need a custom transactor to use DI in this form instead of the "normal" one:

```cs
class MyController
{
  readonly ITransactor transactor;

  public MyController(OurCompanyTransactor t) => transactor = t;
}
```

Reply:
>IMO, it's perfectly fine to instruct any team that need a custom transactor to use DI in this form instead of the "normal" one:

This way is very error prone, and might lead to many frustrated application issues. I am not sue if I agree that it's "perfectly fine", but let it be, I guess.

Reply:
Hm, ok. If you think it's that big of a concern, them maybe I can figure something out.

My thinking was that this type of a approach is very general, and outside the scope of Starcounter. You use containment to customize/"override" behavior of the services we define, and then exactly how you do it in your application, that's up to you.

It's that balance again - should we a) keep adding more APIs, or should we b) recommend patterns developers can use to extend our core features to achieve the same thing themselves?

The pros of (a) - keep growing the API - is that it's less work for the developer and that Starcounter become easier to grasp. The cons are that it's more code for us to maintain - test, keep compatible, document - and that it's less dynamic.

Pros and cons of (b) - trying to keep a smaller API, and suggest how it can be extended - is the opposite. One thing I do like about it (except its less maintenance) is that it become very clear what the very heart of the product is.